src/middlewares/validator.js

```diff
@@ -0,0 +1,31 @@
+const createError = require('http-errors')
+const Ajv = require('ajv')
+const ajvKeywords = require('ajv-keywords')
+
+const ajv = new Ajv({v5: true, $data: true, allErrors: true})
+ajvKeywords(ajv)
+
+module.exports = (inputSchema, outputSchema, options) => ({
+  before: (handler, next) => {
+    if (!inputSchema) {
+      return next()
+    }
+    const validate = ajv.compile(inputSchema)
```

Comment:
Would it make more sense, for performance, to pre-compile the schema, outside the `before` function (so that we can reuse across multiple calls)